documentation/src/main/wiki/BestPractices.md

```diff
@@ -0,0 +1,218 @@
+	<screen/>
+</test>
+```
+The test doesn't click the button unless it's visible and clickable. The maximum waiting time is 0ms, 
+so unless the button is ready right away, the click command is ignored, 
+an exception is thrown in the logs and the test continues.
+
+#### Suplement wait commands with sleep commands
+
+Using wait commands and commands with timeouts is good for interacting with specific elements of the page under test. 
+Using the sleep command after that is even better, because it allows the page under test to load 
+without having the user write a wait command for each element on the page that might still be loading.
+
+##### Example
+```xml
+<test>
+	[...]
+	<click css="#bigButton" timeout="5000"/>
+	<sleep duration="5000"/>
+	<screen/>
+</test>
+```
+The screenshot taken during this test captures how the page looks like five seconds after the #bigButton is clicked. 
+First the button is clicked if it appears within five seconds, then the test waits five seconds for the page to load 
+and eventually the screenshot is taken.
+
+#### Don't overuse the sleep command
+
+Don't rely solely on sleep commands with huge durations. Using wait commands instead of sleep allows the tests 
+to run much faster in case if pages load properly and aren't broken.
+
+Examples
+
+###### Script A
+```xml
+<test>
+	[...]
+	<sleep duration="20000"/>
+	<screen/>
+</test>
+```
+
+This test test waits 20 seconds no matter how fast the page and the elements we're interested in load.
+
+###### Script B
+```xml
+<test>
+	[...]
+	<click css="#bestButton" timeout="15000"/>
+	<sleep duration="5000"/>
+	<screen/>
+</test>
+```
+
+This test might wait 20 seconds if the page loads slowly or fails to work properly, 
+but if everything is OK then the test might take even 5 seconds.
+
+#### Hide or replace unstable elements
+
+Some page elements such as images or third-party components might not load as fast as the rest of the page under test. 
+In cases where we don't really need to test this slowly loading element we can either hide it or replace a large, 
+heavy image with a small placeholder.
+
+Example
+
+```xml
+<test>
+	[...]
+	<hide css="#uselessElement" leaveBlankSpace="false" timeout="5000"/>
+	<replaceText css="img" attributeName="src" value="/content/dam/test/uk/en_gb/dummy/dummy.png.renditions.original.png" />
+	<replaceText attributeName="style"
+	    value="background-image: url('/content/dam/test/uk/en_gb/dummy/dummy.png.renditions.original.png')"
+	    css="#backgroundStuff" timeout="5000"/>
+	[...]
+	<screen/>
+</test>
+```
+The first command removes the selected element from the page. The second command replaces the src attribute for all img 
+elements on the page under test. The third command replaces the background image for the selected element.
+
+#### When everything else fails - there's JavaScript
+
+Sometimes it can be problematic to get the page under test in a stable state to get always the same screenshot.
```

Comment:
Let's link directly to the modifiers, like `[[hide|HideModifier]]`, `[[executejavascript|ExecuteJavaScriptModifier]]` etc.

Reply:
I'm moving modifier-specific tips to the modifier pages, so this info won't be needed then.